src/config/logger.ts

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable no-console */
+/**
+ * Log levels supported by the logger
+ */
+export enum LogLevel {
+  DEBUG = "debug",
+  INFO = "info",
+  WARN = "warn",
+  ERROR = "error",
+}
+
+/**
+ * Interface for logger adapters
+ */
+export interface LoggerAdapter {
+  debug: (message: string, ...args: unknown[]) => void;
+  info: (message: string, ...args: unknown[]) => void;
+  warn: (message: string, ...args: unknown[]) => void;
+  error: (message: string, ...args: unknown[]) => void;
+}
+
+/**
+ * Default console adapter implementation
+ */
+const consoleAdapter: LoggerAdapter = {
+  debug: (message: string, ...args: unknown[]) => {
+    if (__DEV__) {
+      console.debug(`[DEBUG] ${message}`, ...args);
```

Comment:
maybe we could use our existing `debug()` helper to pretty print debug logs here with timestamps, what do you think?